tlslite/utils/python_tripledes.py

```diff
@@ -0,0 +1,497 @@
+
+# Author:   Todd Whiteman
+# Date:     16th March, 2009
+# Verion:   2.0.0
+# License:  Public Domain - free to do as you wish
+# Homepage: http://twhiteman.netfirms.com/Des.html
+#
+# Modified by: Adam Varga, 2018
+#
+# A pure python implementation of the DES and Triple DES
+# encryption algorithms using CBC mode. Triple DES class is
+# implemented by utilising the DES base. Triple DES is
+# DES-EDE3 with a 24 byte key, or DES-EDE2 with a 16 byte key.
+
+"""
+Class initialization
+--------------------
+pyDes.Des(key, iv)
+pyDes.Python_TripleDES(key, iv)
+
+key -> Bytes containing the encryption key. 8 bytes for DES, 16 or 24 bytes
+       for Triple DES
+iv  -> Initialization Vector in bytes. Length must be 8 bytes.
+"""
+
+import sys
+import warnings
+
+# PY_VER is used to handle Python2 and Python3 differences.
+PY_VER = sys.version_info
+FLAG = 1
+
+def new(key, iv):
+    # the new() method to operate this cipher
+    return Python_TripleDES(key, iv)
+
+
+class _baseDes(object):
+    """ The base class shared by des and triple des.
+    """
+    def __init__(self, iv=None):
+        if iv:
+            iv = self._guard_against_unicode(iv)
+        self.block_size = 8
+
+        # Sanity checking of arguments.
+        if not iv:
+            raise ValueError("Initialization Vector (iv) must be supplied")
+        if iv and len(iv) != self.block_size:
+            raise ValueError("Invalid Initialization Vector (iv), must be a "
+                             "multiple of {0} bytes".format(self.block_size))
+        self._iv = iv
+
+    @property
+    def iv(self):
+        # return bytes
+        return self._iv
+
+    @iv.setter
+    def iv(self, iv):
+        self._iv = iv
+
+    def _guard_against_unicode(self, data):
+        # Only accept byte strings or ascii unicode values, otherwise
+        # there is no way to correctly decode the data into bytes.
+
+        if PY_VER < (3, ):
+            if isinstance(data, str):
+                warnings.warn("Only bytes, bytearray or memoryview "
+                              "objects of them should be passed",
+                              DeprecationWarning,
+                              stacklevel=1)
+            if isinstance(data, unicode):
+                raise ValueError("Only bytes, bytearray or memoryview "
+                                 "objects of them should be passed, "
+                                 "not Unicode strings")
+        else:
+            if isinstance(data, str):
+                warnings.warn("Only bytes, bytearray or memoryview "
+                              "objects of them should be passed",
+                              DeprecationWarning,
+                              stacklevel=1)
+                # Only accept ascii unicode values.
+                try:
+                    return data.encode('ascii')
+                except UnicodeEncodeError:
+                    pass
+                raise ValueError("The Unicode strings shouldn't be passed")
```

Comment:
that's a weird construction... why `pass` in `except` and then right away raise exception?